### PR TITLE
[feat-4996] hoofdcategorie tegel

### PR DESCRIPTION
**Pull request by vdegraaf**
Ticket: [SIG-4996](https://gemeente-amsterdam.atlassian.net/browse/SIG-4996)

### Context
Adding main categories to the configurable settings in the backoffice. 

### Changes
- Add and rename routes
- Make form and detail generic components
- Implement for subcategories
- Implement for main categories
- Add and adjust tests

N.B. 
changes for a different PR
-  Upgrade detail test
-  Make images available in main categories overview

### CODE REVIEW

**Comment by OscarBakker** [comment]
When I reload the page is see this in an instant, although I have permission

Screenshot 2023-04-25 at 11 11 37

**Review by SireeKoolenWijkstra — COMMENTED** [review]
- [ ] When I click on Afval I get directed to an empty page.
- [ ] It says there is no icon on Wegen, verkeer, straatmeubilair, but you can see in it's history that there is one. Is this for a later date PR?

File: internals/schemas/app.schema.json [hunk 1]
```diff
@@ -145,6 +145,10 @@
         "showDashboard": {
           "description": "Shows the realtimedashboard",
           "type": "boolean"
+        },
+        "showMainCategories": {
+          "description": "Shows main categories in settings.",
```
[the remark this answers is not in the document]
Comment (vdegraaf) [reply internals/schemas/app.schema.json:150]: Poteto potato :p half of them have it, the other half not. Removed it for you;)

File: src/signals/settings/categories/components/CategoryForm.test.tsx [hunk 2]
```diff
@@ -73,5 +76,55 @@ describe('CategoryForm', () => {
...
+
+    expect(
+      screen.queryByText('Verantwoordelijke afdeling')
+    ).not.toBeInTheDocument()
+  })
+
+  it('should not be able to change public name when is_public_accessible', () => {
+    const mockFormValuesNotAccessible = {
+      ...mockFormValues,
+      is_public_accessible: false,
```
[the remark this answers is not in the document]
Comment (vdegraaf) [reply src/signals/settings/categories/components/CategoryForm.test.tsx:115]: jup

**Comment by vdegraaf** [comment]
> 

This is the same as in acceptance currently

**Comment by vdegraaf** [comment]
> * [ ]  When I click on Afval I get directed to an empty page.
> * [ ]  It says there is no icon on Wegen, verkeer, straatmeubilair, but you can see in it's history that there is one. Is this for a later date PR?

1) Good point. No idea. I asked the backend to take a look. It's because the history endpoint returns a 500

2) If you look in django it doesn't have an icon. So this is correct. Not sure why the history says so.

**Review by SireeKoolenWijkstra — APPROVED** [review]
lgtm

**Review by OscarBakker — APPROVED** [review]
good

**Pull request merged**